[CORE-8436]: Add support for SASL/PLAIN

michael-redpanda opened the pull request:
<!--
See https://github.com/redpanda-data/redpanda/blob/dev/CONTRIBUTING.md#pull-request-body
for more details and examples of what is expected in a PR body.

Content in this top section is REQUIRED. Describe, in plain language, the motivation
behind the change (bug fix, feature, improvement) in this PR and how the included
commits address it.

Add the GitHub keyword `Fixes` to link to bug(s) this PR will fix, e.g.
  Fixes #ISSUE-NUMBER, Fixes #ISSUE-NUMBER, ...

If this PR is a backport, link to the original with `Backport of PR`, e.g.
  Backport of PR #PR-NUMBER
-->

Adds support for SASL/PLAIN authentication mechanism.

*Configuration*

Adds `"PLAIN"` as a valid entry in the `sasl_mechanisms` cluster config list.

*Caveats*

`"PLAIN"` cannot be the only entry in the `sasl_mechanisms` list.  Redpanda will log either at `warn` when `"PLAIN"` is enabled.  Will log at `err` if `"PLAIN"` is enabled and TLS is not enabled on any interface.

Fixes: CORE-8436
Fixes: CORE-8458
Fixes: CORE-8459
Fixes: CORE-8460


Todo:

- [x] Test for error and warning logs
- [x] Validate `"PLAIN"` only is an error

## Backports Required

<!-- Checking at least one of the checkboxes is REQUIRED if this PR is not a backport. -->

- [X] none - not a bug fix
- [ ] none - this is a backport
- [ ] none - issue does not exist in previous branches
- [ ] none - papercut/not impactful enough to backport
- [ ] v24.3.x
- [ ] v24.2.x
- [ ] v24.1.x

## Release Notes

<!--
If the changes in this PR do not need to be mentioned in the release
notes, then don't add a sub-section and simply list `none`, e.g.

* none

Otherwise, adding a sub-section or `none` is REQUIRED if the PR is not a backport PR.
If this is a backport PR, adding contents to this section will override
the release notes section inherited from the original PR to dev.

Add one or more of the sub-sections with a short description bullet
point of the change, e.g.

### Bug Fixes

* Short description of the bug fix if this is a PR to `dev` branch.

### Features

* Short description of the feature. Explain how to configure.

### Improvements

* Short description of how this PR improves existing behavior.

-->

### Features

* Redpanda now supports the SASL/PLAIN authentication mechanism.  To enable it add `"PLAIN"` to the `sasl_mechanisms` cluster configuration list

BenPope committed:
security: Introduce plain_authenticator

SUCCESS-02

Signed-off-by: Ben Pope <ben@redpanda.com>

BenPope committed:
security: Extract validate_scram_credential

Pure refactor, no change in behaviour.

Signed-off-by: Ben Pope <ben@redpanda.com>

BenPope committed:
security: Improve validate_scram_credential

Optimise the password validation when SCRAM-SHA-512 is in use,
by avoiding the validation against SCRAM-SHA-256 that will fail.

Signed-off-by: Ben Pope <ben@redpanda.com>

BenPope committed:
feature_manager: Log bad SASL/PLAIN config

REQ-02

Signed-off-by: Ben Pope <ben@redpanda.com>

BenPope committed:
config: Add configuration for SASL/PLAIN

Signed-off-by: Ben Pope <ben@redpanda.com>

BenPope committed:
kafka: Wire up SASL/PLAIN

Signed-off-by: Ben Pope <ben@redpanda.com>

BenPope committed:
feature_manager: Refactor any_of from absl to std::ranges

No change in behaviour

Signed-off-by: Ben Pope <ben@redpanda.com>

michael-redpanda committed:
dt: Added 'PLAIN' to list of simple SASL mechanisms

This change allows for the selection of SASL/PLAIN as an authentication
method using the Python RdKafka library.

Signed-off-by: Michael Boquard <michael@redpanda.com>

michael-redpanda committed:
dt: Added test to exercise SASL/PLAIN using various clients

Added SASL/PLAIN verification tests for:
* KCL
* RPK
* Python RdKafka
* Kafka CLI tools

Validated that both SCRAM-256 and SCRAM-512 users worked.

Fixes: CORE-8458
Fixes: CORE-8459
Fixes: CORE-8460

Tested that a log message appears when SASL/PLAIN is enabled and that
PLAIN cannot be the only SASL mechanism in the mechanisms list.

Signed-off-by: Michael Boquard <michael@redpanda.com>

michael-redpanda committed:
dt: Permitted selection of SASL algorithm with kafka cli

Signed-off-by: Michael Boquard <michael@redpanda.com>

michael-redpanda committed:
security/test: Added unit tests for plain authenticator

Signed-off-by: Michael Boquard <michael@redpanda.com>

michael-redpanda commented:
Force push [`ea25a3f`](https://github.com/redpanda-data/redpanda/compare/fe747785a296e3d5afc5627480835f2fa33bf7ac..ea25a3f3fbf0119e161194ed6c97abcfb082828a):

* Adjusted log logic to also log at error if not Kafka API TLS entries are present
* Fixed spelling
* Added tests for log messages and config

michael-redpanda commented on src/v/security/scram_authenticator.cc:
```diff
@@ -137,4 +137,22 @@ scram_authenticator<T>::authenticate(bytes auth_bytes) {
 template class scram_authenticator<scram_sha256>;
 template class scram_authenticator<scram_sha512>;
 
+std::optional<std::string_view> validate_scram_credential(
+  const scram_credential& cred, const credential_password& password) {
+    std::optional<std::string_view> sasl_mechanism;
+    if (
+      cred.stored_key().size() == 32
+      && security::scram_sha256::validate_password(
+        password, cred.stored_key(), cred.salt(), cred.iterations())) {
+        sasl_mechanism = security::scram_sha256_authenticator::name;
+    } else if (
+      cred.stored_key().size() == 64
+      && security::scram_sha512::validate_password(
+        password, cred.stored_key(), cred.salt(), cred.iterations())) {
+        sasl_mechanism = security::scram_sha512_authenticator::name;
+    }
+
+    return sasl_mechanism;
```
nit: Instead of hard coding the values you could do something like:

```suggestion
std::optional<std::string_view> validate_scram_credential(
  const scram_credential& cred, const credential_password& password) {
    std::optional<std::string_view> sasl_mechanism;
    if (
      cred.stored_key().size() == crypto::digest_ctx::size(crypto::digest_type::SHA256)
      && security::scram_sha256::validate_password(
        password, cred.stored_key(), cred.salt(), cred.iterations())) {
        sasl_mechanism = security::scram_sha256_authenticator::name;
    } else if (
      cred.stored_key().size() == crypto::digest_ctx::size(crypto::digest_type::SHA512)
      && security::scram_sha512::validate_password(
        password, cred.stored_key(), cred.salt(), cred.iterations())) {
        sasl_mechanism = security::scram_sha512_authenticator::name;
    }

    return sasl_mechanism;
```

michael-redpanda replied:
@pgellert 's solution is nicer :)

michael-redpanda commented on src/v/kafka/server/server.cc:
```diff
@@ -720,6 +721,16 @@ ss::future<response_ptr> sasl_handshake_handler::handle(
         }
     }
 
+    if (supports("PLAIN")) {
+        supported_sasl_mechanisms.emplace_back(
+          security::plain_authenticator::name);
+        if (request.data.mechanism == security::plain_authenticator::name) {
+            ctx.sasl()->set_mechanism(
+              std::make_unique<security::plain_authenticator>(
+                ctx.credentials()));
+        }
+    }
```
discussion: Getting clarification from REQ-01.  Not sure if the wording was wrong or if `PLAIN` implies also enabling `SCRAM`.

pgellert replied:
Is it "{PLAIN} implies {PLAIN, SCRAM}" or "{PLAIN} without SCRAM is invalid" or "{PLAIN} implies only {PLAIN} on the kafka procotol, but you can still manage SCRAM users through the admin API"?

BenPope replied:
PLAIN means enable PLAIN. One other secure mechanism must be explicitly enabled.

michael-redpanda commented:
Force push [`7771b04`](https://github.com/redpanda-data/redpanda/compare/ea25a3f3fbf0119e161194ed6c97abcfb082828a..7771b0433bfba6b434579c6b5712ff33dbdb5faf):

* Fixed cpplint issue

vbotbuildovich commented:
<!--NEW FAILURES-->
<!--6a58d844-15db-494c-bc46-581362e683e4-->
<!--NEW FAILURES-->
<!--6c8ff3f3-1330-4a76-b4e3-61d3f152cdba-->
<!--body-->
non flaky failures in https://buildkite.com/redpanda/redpanda/builds/59603#0193b6b2-1bf2-4981-b99e-25425073658b:
```
"rptest.tests.e2e_shadow_indexing_test.EndToEndHydrationTimeoutTest.test_hydration_completes_when_consumer_killed"
```
non flaky failures in https://buildkite.com/redpanda/redpanda/builds/59603#0193b6b2-1bf0-4bbb-97af-594a8376f0df:
```
"rptest.tests.simple_e2e_test.SimpleEndToEndTest.test_relaxed_acks.write_caching=False"
```
<!--/body-->

vbotbuildovich commented:
<!--FAILURES SLASH COMMAND https://buildkite.com/redpanda/redpanda/builds/59603-->
<!--6af78ff3-b3dc-4006-8319-6f89b79c611d-->
<!--36479602-c462-4239-8491-082fd2a74584-->
<!--title-->
## Retry command for [Build#59603](https://buildkite.com/redpanda/redpanda/builds/59603)
please wait until all jobs are finished before running the slash command
<!--/title-->
<!--body-->
```
/ci-repeat 1
tests/rptest/tests/simple_e2e_test.py::SimpleEndToEndTest.test_relaxed_acks@{"write_caching":false}
tests/rptest/tests/e2e_shadow_indexing_test.py::EndToEndHydrationTimeoutTest.test_hydration_completes_when_consumer_killed
```
<!--/body-->

vbotbuildovich commented:
<!--RETRIED FAILURES-->
<!--bf2c4195-73e0-4599-82bb-3ddd963ac4b5-->
<!--RETRIED FAILURES-->
<!--95d6a010-0240-4929-879b-5a7b81baaf22-->
<!--body-->
ducktape was retried in https://buildkite.com/redpanda/redpanda/builds/59603#0193b6b2-1bf2-4981-b99e-25425073658b
ducktape was retried in https://buildkite.com/redpanda/redpanda/builds/59603#0193b6b2-1bf1-4910-8193-47fff943528e
<!--/body-->

vbotbuildovich commented:
<!--UNIT FAILURES-->
<!--c2053f20-d70b-49b2-820b-4993591b3da0-->
<!--body-->
the below tests from https://buildkite.com/redpanda/redpanda/builds/59656#0193bb20-bbaa-4494-87a5-ce21272fd252 have failed and will be retried
```
gtest_raft_rpunit
```
<!--/body-->

BenPope commented on tests/rptest/clients/kafka_cli_tools.py:
```diff
@@ -72,6 +72,7 @@ def __init__(self,
                  version: str | None = None,
                  user: str | None = None,
                  passwd: str | None = None,
+                 algorithm: str | None = None,
```
Did adding this not break any call-sites? I'd have kept the `'SCRAM-SHA-256'` default.

BenPope commented on src/v/security/tests/plain_authenticator_test.cc:
```diff
@@ -0,0 +1,242 @@
+      const ss::sstring& username, const ss::sstring& password) {
+        bytes rv;
+        rv.push_back('\0');
+        if (!username.empty()) {
+            auto username_bytes = bytes::from_string(username);
+            std::copy(
+              std::make_move_iterator(username_bytes.begin()),
+              std::make_move_iterator(username_bytes.end()),
+              std::back_inserter(rv));
+        }
+        rv.push_back('\0');
+        if (!password.empty()) {
+            auto password_bytes = bytes::from_string(password);
+            std::copy(
+              std::make_move_iterator(password_bytes.begin()),
+              std::make_move_iterator(password_bytes.end()),
+              std::back_inserter(rv));
+        }
```
Nitpick:
```suggestion
        std::ranges::move(bytes::from_string(username), std::back_inserter(rv));
        rv.push_back('\0');
        std::ranges::move(bytes::from_string(password), std::back_inserter(rv));
```

BenPope commented on src/v/security/tests/plain_authenticator_test.cc:
```diff
@@ -0,0 +1,242 @@
+    ss::sstring password("password");
+    auto authn_msg = bytes::from_string("authz");
+    auto authn_msg_tmp = create_authn_message(username, password);
+    std::copy(
+      std::make_move_iterator(authn_msg_tmp.begin()),
+      std::make_move_iterator(authn_msg_tmp.end()),
+      std::back_inserter(authn_msg));
```
```suggestion
    std::ranges::move(authn_msg_tmp, std::back_inserter(authn_msg));
```

BenPope commented on src/v/security/tests/plain_authenticator_test.cc:
```diff
@@ -0,0 +1,242 @@
+    EXPECT_EQ(res.assume_error(), security::errc::invalid_credentials);
+    EXPECT_FALSE(authn->complete());
+    EXPECT_TRUE(authn->failed());
+}
```
Possibly missing a test for invalid utf-8

michael-redpanda replied:
you're an invalid utf8

BenPope replied:
Fair, but the character is not rendered or pronounced, so nobody has noticed yet.

pgellert commented on src/v/security/scram_authenticator.cc:
```diff
@@ -140,11 +140,15 @@ template class scram_authenticator<scram_sha512>;
-    if (security::scram_sha256::validate_password(
-          password, cred.stored_key(), cred.salt(), cred.iterations())) {
+    if (
+      cred.stored_key().size() == 32
```
nit: magic number. We could have:
```suggestion
      cred.stored_key().size() == security::scram_sha256::key_size
```

and then we can define it in terms of the hash digest size.
```
static constexpr size_t key_size = HashType::digest_size;
```

pgellert commented on src/v/config/validators.cc:
```diff
@@ -124,6 +124,12 @@ validate_sasl_mechanisms(const std::vector<ss::sstring>& mechanisms) {
         }
     }
+
+    if (mechanisms.size() == 1 && mechanisms[0] == "PLAIN") {
```
Is the condition supposed to be "at least one other" or "at least SCRAM as well"?

BenPope replied:
> Is the condition supposed to be "at least one other" or "at least SCRAM as well"?

From SUCCESS-02: "We will not support a cluster configured with only a PLAIN mechanism", which I understand to mean that at least one other secure mechanism is required, which includes all of the others; `SCRAM`,`OAUTHBEARER`, `GSSAPI`.

pgellert replied:
The preceding sentence of SUCCESS-02 is "Users only need to create a single account to support either SCRAM or PLAIN authentication." which makes me think that it is SCRAM specifically that's also required. I think it's worth asking @deniscoady to see what he meant here. (I'd comment on the PRD, but I don't have access.)

BenPope replied:
> "Users only need to create a single account to support either SCRAM or PLAIN authentication.",

SCRAM Users are the single source of truth for users which are managed internally. Historically I've pushed for them to be named "SCRAM Uses"  over "SASL Users" (c.f., SASL/OAUTHBEARER, SASL/GSSAPI, which are managed externally), but although they are created as SCRAM users, they are supported via HTTP Basic Auth, and now, via SASL/PLAIN.

pgellert replied:
Right, I've got that. I guess what I'm trying to say is that it's not clear to me is whether the goal is:
(a) For Redpanda to be opinionated and nudge customers to enable at least one secure sasl mechanism, or
(b) For Redpanda to signal that SCRAM users and SASL users are the same so you probably want to enable SCRAM as well if you enable PLAIN.

If (a), then this implementation makes sense. But I think (b) would also be reasonable, given that I would expect most customers to use PLAIN while migrating over to Redpanda and away from PLAIN over to SCRAM. In that case, I think they would start by enabling PLAIN+SCRAM, then migrate, and finally disable PLAIN.

michael-redpanda replied:
I discussed with @deniscoady and I think we may modify this requirement slightly to make it so if you enable `PLAIN` you _must_ also enable `SCRAM`.  This impacts our internal clients (PP/SR/Auditing) so I think this is reasonable.

pgellert commented on src/v/cluster/feature_manager.cc:
```diff
@@ -310,6 +312,24 @@ ss::future<> feature_manager::maybe_log_license_check_info() {
         }
     }
+
+    if (std::ranges::any_of(
```
nit: it would be nice to make this method less license-specific now that we also log auth-related log messages as well.

BenPope replied:
> nit: it would be nice to make this method less license-specific now that we also log auth-related log messages as well.

@pgellert I'm not sure what the suggestion is here, I think I'm missing some context.

pgellert replied:
This code is inside the method `feature_manager::maybe_log_license_check_info` which has been used specifically for logging the license nag, whereas the new PLAIN-specific error message is unrelated.

My suggestion here is to make the method's name more generic (e.g., `log_periodic_reminders`), extract the part that's license-specific, extract the new code to a new method that has auth/plain in the name, and rename the env-var to something not license-specific (e.g., `__REDPANDA_PERIODIC_REMINDER_INTERVAL_SEC`).

michael-redpanda replied:
@pgellert is your comment that we have some non-license related messages in an area that's normally used just for license related messages?

pgellert replied:
Exactly

pgellert commented on src/v/security/tests/plain_authenticator_test.cc:
```diff
@@ -0,0 +1,242 @@
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0
+
+// #define BOOST_TEST_MODULE kafka_security
```
```suggestion
```

pgellert commented on src/v/security/tests/plain_authenticator_test.cc:
```diff
@@ -0,0 +1,242 @@
+    EXPECT_EQ(res.assume_error(), security::errc::invalid_credentials);
+    EXPECT_FALSE(authn->complete());
+    EXPECT_TRUE(authn->failed());
+}
```
Should we add a test that makes some assertions on the audit user as well?

michael-redpanda replied:
@pgellert what are you looking for irt to those?

pgellert replied:
Just a simple test that verifies that the audit user's name is properly set on success. If you think it's too granular, I'm fine with that, but it would be a sanity check to ensure that the audit user is updated.

pgellert reviewed (commented):
looks good

michael-redpanda committed:
fm/dt: Renamed __REDPANDA_LICENSE_CHECK_INTERVAL_SEC

Renamed `__REDPANDA_LICENSE_CHECK_INTERVAL_SEC` with
`__REDPANDA_PERIODIC_REMINDER_INTERVAL_SEC` to better reflect what
messages this environmental variable is controlling.

Signed-off-by: Michael Boquard <michael@redpanda.com>

michael-redpanda commented:
Force push [`f051233`](https://github.com/redpanda-data/redpanda/compare/7ac13cab06eaa8a2747793ae36f569e4225c9aa0..f051233127caf75b2ee22dc5d8f2a11ad4621226)

* Updated per PR comments

vbotbuildovich commented:
<!--FAILURES SLASH COMMAND https://buildkite.com/redpanda/redpanda/builds/59821-->
<!--52f97357-c1ee-45ad-b92f-4d93f7de6a81-->
<!--9bc7788b-1512-49fc-80de-0203b917b490-->
<!--a8096e3a-6e2c-434a-a8f1-faf56c78a175-->
<!--0be3b5df-0feb-49d0-9ee7-e04f50e70048-->
<!--title-->
## Retry command for [Build#59821](https://buildkite.com/redpanda/redpanda/builds/59821)
please wait until all jobs are finished before running the slash command
<!--/title-->
<!--body-->
```


/ci-repeat 1
tests/rptest/tests/pandaproxy_test.py::PandaProxySASLTest.test_list_topics
tests/rptest/tests/schema_registry_test.py::SchemaRegistryLicenseTest.test_license_nag@{"mode":"redpanda"}
tests/rptest/tests/scram_test.py::SaslPlainConfigTest.test_sasl_plain_log@{"enable_tls":true}
tests/rptest/tests/rbac_test.py::RBACLicenseTest.test_license_nag
tests/rptest/tests/redpanda_oauth_test.py::OIDCLicenseTest.test_license_nag@{"authn_config":{"http_authentication":["OIDC","BASIC"]}}
tests/rptest/tests/redpanda_oauth_test.py::OIDCLicenseTest.test_license_nag@{"authn_config":{"sasl_mechanisms":["OAUTHBEARER","SCRAM"]}}
tests/rptest/tests/redpanda_kerberos_test.py::RedpandaKerberosLicenseTest.test_license_nag
tests/rptest/tests/schema_registry_test.py::SchemaRegistryLicenseTest.test_license_nag@{"mode":"compat"}
tests/rptest/tests/scram_test.py::SaslPlainConfigTest.test_sasl_plain_log@{"enable_tls":false}
```
<!--/body-->

vbotbuildovich commented:
<!--TEST RESULTS-->
<!--70c90412-b6ab-42f3-8332-4a67a9e77c91-->
<!--TEST RESULTS-->
<!--ef7a6f41-d35e-4241-97c9-b673b6e04702-->
<!--TEST RESULTS-->
<!--90b4d5f9-1b37-437e-94c8-c42f3c51a51d-->
<!--title-->
## CI test results
<!--/title-->
<!--body-->
<details>
<summary>test results on <a href="https://buildkite.com/redpanda/redpanda/builds/59821">build#59821</a></summary>


| test_id | test_kind | job_url | test_status | passed |
|:------------------------------------------------------------------------------------------------------------------|:------------|:------------------------------------------------------------------------------------------|:--------------|:---------|
| rptest.tests.pandaproxy_test.PandaProxySASLTest.test_list_topics | ducktape | https://buildkite.com/redpanda/redpanda/builds/59821#0193d177-8010-441c-bdca-2cb75d21d08e | FAIL | 0/1 |
| rptest.tests.rbac_test.RBACLicenseTest.test_license_nag | ducktape | https://buildkite.com/redpanda/redpanda/builds/59821#0193d177-800d-4d5f-88e0-9700ddd19db7 | FAIL | 0/1 |
| rptest.tests.redpanda_kerberos_test.RedpandaKerberosLicenseTest.test_license_nag | ducktape | https://buildkite.com/redpanda/redpanda/builds/59821#0193d177-800f-41cb-af9d-780d73f2edf7 | FAIL | 0/1 |
| rptest.tests.redpanda_oauth_test.OIDCLicenseTest.test_license_nag.authn_config=.http_authentication.OIDC.BASIC | ducktape | https://buildkite.com/redpanda/redpanda/builds/59821#0193d177-800d-4d5f-88e0-9700ddd19db7 | FAIL | 0/1 |
| rptest.tests.redpanda_oauth_test.OIDCLicenseTest.test_license_nag.authn_config=.sasl_mechanisms.OAUTHBEARER.SCRAM | ducktape | https://buildkite.com/redpanda/redpanda/builds/59821#0193d177-800e-40cb-8135-099096669ffb | FAIL | 0/1 |
| rptest.tests.schema_registry_test.SchemaRegistryLicenseTest.test_license_nag.mode=SchemaIdValidationMode.COMPAT | ducktape | https://buildkite.com/redpanda/redpanda/builds/59821#0193d177-800f-41cb-af9d-780d73f2edf7 | FAIL | 0/1 |
| rptest.tests.schema_registry_test.SchemaRegistryLicenseTest.test_license_nag.mode=SchemaIdValidationMode.REDPANDA | ducktape | https://buildkite.com/redpanda/redpanda/builds/59821#0193d177-8010-441c-bdca-2cb75d21d08e | FAIL | 0/1 |
| rptest.tests.scram_test.SaslPlainConfigTest.test_sasl_plain_log.enable_tls=False | ducktape | https://buildkite.com/redpanda/redpanda/builds/59821#0193d177-800f-41cb-af9d-780d73f2edf7 | FAIL | 0/1 |
| rptest.tests.scram_test.SaslPlainConfigTest.test_sasl_plain_log.enable_tls=True | ducktape | https://buildkite.com/redpanda/redpanda/builds/59821#0193d177-8010-441c-bdca-2cb75d21d08e | FAIL | 0/1 |
</details>
<details>
<summary>test results on <a href="https://buildkite.com/redpanda/redpanda/builds/59923">build#59923</a></summary>


| test_id | test_kind | job_url | test_status | passed |
|:-------------------------------------------------------------------------------------------------------------------------------------------|:------------|:------------------------------------------------------------------------------------------|:--------------|:---------|
| rptest.tests.cloud_retention_test.CloudRetentionTest.test_cloud_retention.max_consume_rate_mb=None.cloud_storage_type=CloudStorageType.ABS | ducktape | https://buildkite.com/redpanda/redpanda/builds/59923#0193db35-56d5-49e3-97d4-f2a784954cfd | FAIL | 0/6 |
| rptest.tests.pandaproxy_test.PandaProxySASLTest.test_list_topics | ducktape | https://buildkite.com/redpanda/redpanda/builds/59923#0193db33-6fee-4267-b448-e0359d266ccd | FAIL | 0/1 |
| rptest.tests.pandaproxy_test.PandaProxySASLTest.test_list_topics | ducktape | https://buildkite.com/redpanda/redpanda/builds/59923#0193db35-56d3-4a99-b8bb-ea09805fe11b | FAIL | 0/1 |
</details>
<details>
<summary>test results on <a href="https://buildkite.com/redpanda/redpanda/builds/59957">build#59957</a></summary>


| test_id                                                                                                                                    | test_kind   | job_url                                                                                   | test_status   | passed   |
|:-------------------------------------------------------------------------------------------------------------------------------------------|:------------|:------------------------------------------------------------------------------------------|:--------------|:---------|
| rptest.tests.cloud_retention_test.CloudRetentionTest.test_cloud_retention.max_consume_rate_mb=None.cloud_storage_type=CloudStorageType.ABS | ducktape    | https://buildkite.com/redpanda/redpanda/builds/59957#0193df5b-5ab0-44f8-8819-29489172faf3 | FAIL          | 0/6      |
| rptest.tests.e2e_shadow_indexing_test.EndToEndShadowIndexingTest.test_write.cloud_storage_type=CloudStorageType.ABS                        | ducktape    | https://buildkite.com/redpanda/redpanda/builds/59957#0193df5b-5ab0-44f8-8819-29489172faf3 | FLAKY         | 2/6      |
| rptest.tests.e2e_shadow_indexing_test.EndToEndShadowIndexingTestCompactedTopic.test_write.cloud_storage_type=CloudStorageType.ABS          | ducktape    | https://buildkite.com/redpanda/redpanda/builds/59957#0193df5b-5ab0-44f8-8819-29489172faf3 | FLAKY         | 2/6      |
</details>
<!--/body-->

pgellert reviewed (dismissed):
lgtm (If we do change the requirement discussed in https://github.com/redpanda-data/redpanda/pull/24525/commits/08d63c9224d65c05d8d7f4a94caf137d418c165c#r1887547776 then that will need another change ~~to this PR~~ but otherwise lgtm.)

michael-redpanda commented:
> lgtm (If we do change the requirement discussed in [08d63c9#r1887547776](https://github.com/redpanda-data/redpanda/commit/08d63c9224d65c05d8d7f4a94caf137d418c165c#r1887547776) then that will need another change to this PR but otherwise lgtm.)

That or we will file another PR.  There's discussions on adding metrics and we have plenty of time until release.

BenPope commented on src/v/cluster/feature_manager.cc:
```diff
@@ -274,32 +276,38 @@ feature_manager::report_enterprise_features() const {
               clusterlog.info,
-              "Overriding default license log annoy interval to: {}s",
-              license_check_retry.count());
+              "Overriding default reminder period interval to: {}s",
```
It's probably worth extracting out the duplicate `_license_nag_is_set` functions.

michael-redpanda replied:
Oh I missed changing those as well

BenPope reviewed (commented):
Looks great

michael-redpanda commented:
Force push [`9dec947`](https://github.com/redpanda-data/redpanda/compare/f051233127caf75b2ee22dc5d8f2a11ad4621226..9dec947b97e13462cd1cc60a049d1f9d997d35a7):

* Fixed check for license nag message

michael-redpanda commented:
Force push `98d6056`:

* Fixed python lint issue

vbotbuildovich commented:
<!--FAILURES SLASH COMMAND https://buildkite.com/redpanda/redpanda/builds/59923-->
<!--0d198a3f-01cb-4c5b-8d42-b6cfa1ca79c8-->
<!--b20b1a08-8b4e-46b1-9b2c-80dde2a2e720-->
<!--cec3820d-d597-4c14-ba5f-3195cd49921b-->
<!--title-->
## Retry command for [Build#59923](https://buildkite.com/redpanda/redpanda/builds/59923)
please wait until all jobs are finished before running the slash command
<!--/title-->
<!--body-->
```


/ci-repeat 1
tests/rptest/tests/pandaproxy_test.py::PandaProxySASLTest.test_list_topics
tests/rptest/tests/cloud_retention_test.py::CloudRetentionTest.test_cloud_retention@{"cloud_storage_type":2,"max_consume_rate_mb":null}
```
<!--/body-->

michael-redpanda commented:
Force push `b2d8fa9`:

* Fixed bad spelling of `SCRAM-SHA-256` in `kafka_client`

BenPope reviewed (approved):
Are the todos in the cover letter done?

michael-redpanda commented:
> Are the todos in the cover letter done?

yes!

vbotbuildovich commented:
<!--FAILURES SLASH COMMAND https://buildkite.com/redpanda/redpanda/builds/59957-->
<!--0ea057cf-3437-45b3-b776-aa3e8f5e0031-->
<!--78ce49a6-3a8e-4b53-9046-aa84d91e9d2d-->
<!--title-->
## Retry command for [Build#59957](https://buildkite.com/redpanda/redpanda/builds/59957)
please wait until all jobs are finished before running the slash command
<!--/title-->
<!--body-->
```

/ci-repeat 1
tests/rptest/tests/cloud_retention_test.py::CloudRetentionTest.test_cloud_retention@{"cloud_storage_type":2,"max_consume_rate_mb":null}
```
<!--/body-->

michael-redpanda commented:
CI Failure:
* [CORE-8573](https://redpandadata.atlassian.net/browse/CORE-8573)

[CORE-8573]: https://redpandadata.atlassian.net/browse/CORE-8573?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
